lib/Sema/MiscDiagnostics.cpp

```diff
@@ -451,11 +451,25 @@ static void diagSyntacticUseRestrictions(const Expr *E, const DeclContext *DC,
 
       // Allow for a chain of member_ref exprs that end in a decl_ref expr.
       auto *subExpr = borrowExpr->getSubExpr();
-      while (auto *memberRef = dyn_cast<MemberRefExpr>(subExpr))
-        subExpr = memberRef->getBase();
-      if (!isa<DeclRefExpr>(subExpr)) {
+      while (true) {
+        if (auto *memberRef = dyn_cast<MemberRefExpr>(subExpr)) {
+          subExpr = memberRef->getBase();
+          continue;
+        }
+        if (auto *parenExpr = dyn_cast<ParenExpr>(subExpr)) {
+          subExpr = parenExpr->getSubExpr();
+          continue;
+        }
+        if (auto *borrowExpr = dyn_cast<BorrowExpr>(subExpr)) {
+          subExpr = borrowExpr->getSubExpr();
+          continue;
+        }
```

Comment:
consider `Expr::getSemanticsProvidingExpr`  instead so this also handles try and await.